### PR TITLE
Accept also non-url text

### PR DESCRIPTION
**Pull request by jxw1102**
Currently NearDrop only accepts url type of textMetadata. 
This PR adds support to other type of text by storing them into pasteboard.

### CODE REVIEW

File: NearbyShare/InboundNearbyConnection.swift [hunk 1]
```diff
@@ -22,7 +22,8 @@ class InboundNearbyConnection: NearbyConnection{
 	private var cipherCommitment:Data?
 	
 	private var textPayloadID:Int64=0
-	
+    private var textType:Sharing_Nearby_TextMetadata.TypeEnum = .unknown
```
Comment (sonique6784) [inline NearbyShare/InboundNearbyConnection.swift:25]: indentation

File: NearbyShare/InboundNearbyConnection.swift [hunk 2]
```diff
@@ -119,9 +120,14 @@ class InboundNearbyConnection: NearbyConnection{
...
-				NSWorkspace.shared.open(url)
-			}
+            if let textStr=String(data: payload, encoding: .utf8) {
+                if case .url=textType, let url=URL(string: textStr){
+                    NSWorkspace.shared.open(url)
+                }else{
+                    NSPasteboard.general.clearContents()
+                    NSPasteboard.general.setString(textStr, forType: .string)
+                }
+            }
```
Comment (sonique6784) [inline NearbyShare/InboundNearbyConnection.swift:130]: please review indentation

File: NearbyShare/InboundNearbyConnection.swift [hunk 3]
```diff
@@ -302,15 +308,12 @@ class InboundNearbyConnection: NearbyConnection{
...
-				}
-			}else{
-				rejectTransfer(with: .unsupportedAttachmentType)
-			}
+            let metadata=TransferMetadata(files: [], id: id, pinCode: pinCode, textDescription: meta.textTitle)
+            textPayloadID=meta.payloadID
+            textType=meta.type
+            DispatchQueue.main.async {
+                self.delegate?.obtainUserConsent(for: metadata, from: self.remoteDeviceInfo!, connection: self)
+            }
```
Comment (sonique6784) [inline NearbyShare/InboundNearbyConnection.swift:316]: indentation

**Review by sonique6784 — CHANGES_REQUESTED** [review]

**Comment by PauloJSGG** [comment]
Please merge this, it's super useful @jxw1102  @sonique6784

**Comment by TropicolX** [comment]
@jxw1102 @sonique6784

**Comment by macx** [comment]
Come on, @jxw1102 @sonique6784! I would suggest using Prettier, set up a project standard, save the files and submit the changes to be merged. It should be done within 20 Minutes. But this PR is over one year old. Guys…

**Comment by grishka** [comment]
I do not want this feature in my app.

**Comment by pgamerx** [comment]
> I do not want this feature in my app.

Why??
Everyone wants it

**Comment by grishka** [comment]
Because sending something that doesn't visibly end up anywhere on the receiving side is confusing UX.

AirDrop on Mac does accept text from iPhones but it saves it as a .txt file. I can do this just to get this whole thing over with.

**Comment by PauloJSGG** [comment]
Yes please 😃

**Comment by gramcraka** [comment]
> Because sending something that doesn't visibly end up anywhere on the receiving side is confusing UX.
> 
> AirDrop on Mac does accept text from iPhones but it saves it as a .txt file. I can do this just to get this whole thing over with.

If we present a button that says "Copy" in the notification, that would be a less confusing UX.

**Comment by grishka** [comment]
> If we present a button that says "Copy" in the notification, that would be a less confusing UX.

What if you accidentally dismiss that notification?

**Comment by pgamerx** [comment]
Then you nearby share again

**Comment by flawnn** [comment]
> > If we present a button that says "Copy" in the notification, that would be a less confusing UX.
> 
> What if you accidentally dismiss that notification?

Don't you have that UX Issue already with files you share?

**Comment by macx** [comment]
> Because sending something that doesn't visibly end up anywhere on the receiving side is confusing UX.

So, Apple is wrong in terms of UX by enabling Continuity, which is enabling shared clipboard without any notification? If I copy or send anything, it's by action. The better UX could be, that I trust a device and the files will copy to a specific folder, or a notification button to accept and download it. But I would rather not click a button to accept the text I copied seconds ago – That is a bad UX.

The question I would ask: Do you want to copy a functionality with its limitations or UX fails, or could we do it better? I want to copy 60 % of the clipboard and 40 % of files from Android to Mac. Therefore, currently, this app is 60 % useless for me. But it has potential. Fingers cross to use the original in the future. If not, there a forks.

**Comment by grishka** [comment]
> So, Apple is wrong in terms of UX by enabling Continuity, which is enabling shared clipboard without any notification?

Apple is able to identify which devices are yours and which aren't, with 100% reliability.

**Pull request closed without merging**